### PR TITLE
[CosmosDB] `az cosmosdb delete`: Add `--no-wait`

### PR DESCRIPTION
**Pull request by danielmarbach**
Unfortunately I get a bit stuck trying to figure out how the 'begin_delete' works. Instead of abandoning what I already had I figured I sent in the PR. Maybe the reviewers can point out to me what is left to make this a finished PR.

**Related command**

`az cosmosdb delete`

**Description**

Extends the `az cosmosdb delete` with `--no-wait`

**Testing Guide**

**History Notes**
[CosmosDB] `az cosmosdb delete`: Add no-wait

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
CosmosDB

File: src/azure-cli/azure/cli/command_modules/cosmosdb/commands.py [hunk 1]
```diff
@@ -165,7 +165,7 @@ def load_command_table(self, _):
         g.command('list-connection-strings', 'list_connection_strings', table_transformer=list_connection_strings_output, deprecate_info=g.deprecate(redirect='cosmosdb keys list --type connection-strings', hide=True))
         g.custom_command('regenerate-key', 'cli_cosmosdb_regenerate_key', deprecate_info=g.deprecate(redirect='cosmosdb keys regenerate', hide=True))
         g.command('check-name-exists', 'check_name_exists')
-        g.command('delete', 'begin_delete', confirmation=True)
+        g.command('delete', 'begin_delete', confirmation=True, supports_no_wait=True)
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/cosmosdb/commands.py:168]:
```suggestion
        g.custom_command('delete', 'cli_cosmosdb_delete', confirmation=True, supports_no_wait=True)
```
To support `--no-wait`, you need to add the no wait logic in your custom func (let say in `cli_cosmosdb_delete`), call `client.begin_delete` in `sdk_no_wait` func.

Take this as an example: https://github.com/Azure/azure-cli/blob/168206d11be6b14e692e3b733ad54a261e5832d6/src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py#L2326-L2338

Comment (danielmarbach) [reply src/azure-cli/azure/cli/command_modules/cosmosdb/commands.py:168]: I was searching for the begin delete custom function but couldn't find it. I was aware things should be wrapped but after I couldn't find it I figured begin delete must be some generic function. I'll give it another look

Comment (danielmarbach) [reply src/azure-cli/azure/cli/command_modules/cosmosdb/commands.py]: @evelyn-ys I pushed an update. Please be patient with me. I'm very new to this whole repository and probably doing things horribly wrong. Thanks for your assistance!

Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/cosmosdb/commands.py]: `begin_delete` should be the interface defined in `azure-mgmt-cosmosdb` SDK which Azure CLI cosmosdb module depends on.

If you are new to CLI development, better to read the doc first https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md

File: src/azure-cli/azure/cli/command_modules/cosmosdb/commands.py [hunk 2]
```diff
@@ -165,7 +165,7 @@ def load_command_table(self, _):
         g.command('list-connection-strings', 'list_connection_strings', table_transformer=list_connection_strings_output, deprecate_info=g.deprecate(redirect='cosmosdb keys list --type connection-strings', hide=True))
         g.custom_command('regenerate-key', 'cli_cosmosdb_regenerate_key', deprecate_info=g.deprecate(redirect='cosmosdb keys regenerate', hide=True))
         g.command('check-name-exists', 'check_name_exists')
-        g.command('delete', 'begin_delete', confirmation=True)
+        g.command('delete', 'cli_cosmosdb_delete', confirmation=True, supports_no_wait=True)
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/cosmosdb/commands.py:168]:
```suggestion
        g.custom_command('delete', 'cli_cosmosdb_delete', confirmation=True, supports_no_wait=True)
```

File: src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py [hunk 3]
```diff
@@ -516,6 +516,10 @@ def cli_cosmosdb_update(client,
     docdb_account = client.get(resource_group_name, account_name)  # Workaround
     return docdb_account
 
+def cli_cosmosdb_delete(client, resource_group_name, account_name, no_wait=False):
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py:519]:
```suggestion

def cli_cosmosdb_delete(client, resource_group_name, account_name, no_wait=False):
```

**Review by calvinhzy — APPROVED** [review]

**Pull request merged**